tensorflow/c/experimental/filesystem/modular_filesystem.cc

```diff
@@ -390,6 +390,30 @@ void ModularFileSystem::FlushCaches(TransactionToken* token) {
   if (ops_->flush_caches != nullptr) ops_->flush_caches(filesystem_.get());
 }
 
+Status ModularFileSystem::SetConfiguration(const std::string& name,
+                                           const std::string& value) {
+  if (ops_->set_filesystem_configuration == nullptr) {
+    return errors::Unimplemented(
+        "Filesystem does not support SetConfiguration()");
+  }
+
+  TF_Filesystem_Option option;
+  memset(&option, 0, sizeof(option));
+  option.name = const_cast<char*>(name.c_str());
+  option.per_file = 0;
+  option.type_tag = TF_FILESYSTEM_OPTION_TYPE_BUFFER;
+  option.num_values = 1;
+  TF_Filesystem_Option_Value option_value[1];
+  memset(&option_value, 0, sizeof(option_value));
+  option_value[0].buffer_val.buf = const_cast<char*>(value.c_str());
+  option_value[0].buffer_val.buf_length = value.size();
+  option.value = &option_value[0];
```

Comment:
@yongtang in case of configs that might need multiple values (for ex: connecting to an HDFS cluster where the active/passive namenodes might change), this approach indicates that we provide both the namenodes in a single conf value (for ex: "," delimited) and then handle the connection logic in the filesystem plugin implementation. Please let me know if this understanding is correct. Else, what would be a good approach to handle such cases?

Reply:
@kvignesh1420 It is possible in C API to pass multiple values with one key. Let me see if I can update the PR.

Reply:
sure!

Reply:
@kvignesh1420 The PR has been updated.

Reply:
Thanks @yongtang LGTM.